spec/ibandit/german_details_converter_spec.rb

```diff
@@ -0,0 +1,1429 @@
+require 'spec_helper'
+
+describe Ibandit::GermanDetailsConverter do
+  describe '.convert' do
+    subject(:convert) { described_class.convert(args) }
+    let(:args) { { bank_code: bank_code, account_number: account_number } }
+    let(:account_number) { '1234567890' }
+    let(:bank_code) { '10010010' }
+
+    it 'delegates to the right rule' do
+      expect_any_instance_of(Ibandit::GermanDetailsConverter::Rule000000).
+        to receive(:converted_details).
+        and_call_original
+      convert
+    end
+
+    context 'with an unknown bank code' do
+      let(:bank_code) { 'XXXXXXXX' }
+      it { is_expected.to eq(args) }
+    end
+
+    context 'with rule' do
+      before do
+        allow(described_class).
+          to receive(:rules).
+          and_return(bank_code => { iban_rule: rule })
+      end
+
+      context '000000' do
```

Comment:
As discussed, could probably shorten up these unit tests with something like (roughly tested):

```ruby
require 'spec_helper'

describe 'convertor unit tests' do
  context "JSON fixture" do
    json_fixture('germany_test_cases').each do |convertor|
      context "Rule #{convertor['convertor']}" do
        let(:klass) do
          Ibandit::GermanDetailsConverter.const_get("Rule#{convertor['convertor']}")
        end

        subject { klass.new(bank_code, account_number).converted_details }

        convertor.fetch('valid', []).each do |pair|
          context "bank code: #{pair['bank_code']} account number #{pair['account_number']}" do
            let(:bank_code) { pair['bank_code'] || "00000000" }
            let(:account_number) { pair['account_number'] }
            let(:converted_bank_code) { pair['exp_bank_code'] || bank_code }
            let(:converted_account_number) { pair['exp_account_number'] || account_number }
            it { is_expected.to eq({bank_code: converted_bank_code, account_number: converted_account_number}) }
          end
        end

        convertor.fetch('invalid', []).each do |pair|
          context "bank code: #{pair['bank_code']} account number #{pair['account_number']}" do
            let(:bank_code) { pair['bank_code'] || "00000000" }
            let(:account_number) { pair['account_number'] }
            it 'raises UnsupportedAccountDetails' do
              expect{ subject }.to raise_error(Ibandit::UnsupportedAccountDetails)
            end
          end
        end
      end
    end
  end
end
```

with test data of the form:

```json
[
  {
    "convertor": "000000",
    "valid": [
      { "bank_code": "10010010", "account_number" : "1234567890" }
    ]
  },
  {
    "convertor": "000100",
    "invalid": [
      { "bank_code": "10010010", "account_number" : "1234567890" }
    ]
  },
  {
    "convertor": "000200",
    "valid": [
      { "bank_code": "10010010", "account_number" : "1234567890" }
    ],
    "invalid": [
      { "bank_code": "10010010", "account_number" : "1111111611" },
      { "bank_code": "10010010", "account_number" : "1111111861" }
    ]
  },
  {
    "convertor": "000300",
    "valid": [
      { "bank_code": "10010010", "account_number" : "1234567890" }
    ],
    "invalid": [
      { "bank_code": "10010010", "account_number" : "6161604670" }
    ]
  }
]
```

Not critical though.